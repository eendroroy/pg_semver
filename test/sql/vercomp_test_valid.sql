LOAD 'vercomp';

CREATE TABLE versions(version VERSION);

INSERT INTO versions VALUES ('0');
INSERT INTO versions VALUES ('0.0');
INSERT INTO versions VALUES ('0.1');
INSERT INTO versions VALUES ('0.0.0');
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.0.1-alpha1');
INSERT INTO versions VALUES ('0.0.1-beta1');
INSERT INTO versions VALUES ('0.0.1-rc1');



INSERT INTO versions VALUES ('!0');
INSERT INTO versions VALUES ('@0.0');
INSERT INTO versions VALUES ('0.#1');
INSERT INTO versions VALUES ('0.0.%0');
INSERT INTO versions VALUES ('0.0.1');
INSERT INTO versions VALUES ('0.&0.1-alpha');

SELECT * FROM versions;

DROP TABLE IF EXISTS versions;
