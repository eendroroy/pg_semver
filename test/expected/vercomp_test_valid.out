LOAD 'vercomp';
CREATE TABLE ver_test(version VER);
INSERT INTO ver_test VALUES ('0');
INSERT INTO ver_test VALUES ('0.0');
INSERT INTO ver_test VALUES ('0.1');
INSERT INTO ver_test VALUES ('0.0.0');
INSERT INTO ver_test VALUES ('0.0.1');
INSERT INTO ver_test VALUES ('0.0.1-a');
INSERT INTO ver_test VALUES ('0.0.1-alpha');
INSERT INTO ver_test VALUES ('0.0.1-a.0');
INSERT INTO ver_test VALUES ('0.0.1-a.0.1');
INSERT INTO ver_test VALUES ('0.0.1-alpha.0');
INSERT INTO ver_test VALUES ('0.0.1-alpha.0.1');
INSERT INTO ver_test VALUES ('0.0.1-b');
INSERT INTO ver_test VALUES ('0.0.1-beta');
INSERT INTO ver_test VALUES ('0.0.1-b.0');
INSERT INTO ver_test VALUES ('0.0.1-b.0.1');
INSERT INTO ver_test VALUES ('0.0.1-beta.0');
INSERT INTO ver_test VALUES ('0.0.1-beta.0.1');
INSERT INTO ver_test VALUES ('0.0.1-rc');
INSERT INTO ver_test VALUES ('0.0.1-rc.0');
INSERT INTO ver_test VALUES ('0.0.1-rc.0.1');
INSERT INTO ver_test VALUES ('!0');
ERROR:  Invalid Version !0.
LINE 1: INSERT INTO ver_test VALUES ('!0');
                                     ^
INSERT INTO ver_test VALUES ('@0.0');
ERROR:  Invalid Version @0.0.
LINE 1: INSERT INTO ver_test VALUES ('@0.0');
                                     ^
INSERT INTO ver_test VALUES ('0.#1');
ERROR:  Invalid Version 0.#1.
LINE 1: INSERT INTO ver_test VALUES ('0.#1');
                                     ^
INSERT INTO ver_test VALUES ('0.0.%0');
ERROR:  Invalid Version 0.0.%0.
LINE 1: INSERT INTO ver_test VALUES ('0.0.%0');
                                     ^
INSERT INTO ver_test VALUES ('0.0.1');
INSERT INTO ver_test VALUES ('0.0^.1-a');
ERROR:  Invalid Version 0.0^.1-a.
LINE 1: INSERT INTO ver_test VALUES ('0.0^.1-a');
                                     ^
INSERT INTO ver_test VALUES ('0.&0.1-alpha');
ERROR:  Invalid Version 0.&0.1-alpha.
LINE 1: INSERT INTO ver_test VALUES ('0.&0.1-alpha');
                                     ^
INSERT INTO ver_test VALUES ('0.0.*1-a.0');
ERROR:  Invalid Version 0.0.*1-a.0.
LINE 1: INSERT INTO ver_test VALUES ('0.0.*1-a.0');
                                     ^
SELECT * FROM ver_test;
     version     
-----------------
 0
 0.0
 0.1
 0.0.0
 0.0.1
 0.0.1-a
 0.0.1-alpha
 0.0.1-a.0
 0.0.1-a.0.1
 0.0.1-alpha.0
 0.0.1-alpha.0.1
 0.0.1-b
 0.0.1-beta
 0.0.1-b.0
 0.0.1-b.0.1
 0.0.1-beta.0
 0.0.1-beta.0.1
 0.0.1-rc
 0.0.1-rc.0
 0.0.1-rc.0.1
 0.0.1
(21 rows)

DROP TABLE IF EXISTS ver_test;
